### PR TITLE
Model integration and documentation

### LINKED ISSUE #27: Integrate Our Hugging Face Model in App and Document the findings
Since our existing UI already has an existing CodeT5 hugging face model integrated to it, replace that one with our trained model and look for any bugs or issues

Acceptance criteria:
- Replace our trained hugging face model with the existing one 
- Look out for any bugs or issues
- Document about the model training , hyperparameter tuning and model evaluation for training on more data for future.

### PR DESCRIPTION
**Pull request by oam67**
Fixes #27

**What was changed?**

I replaced the previously integrated HuggingFace CodeT5 model with our newly fine-tuned model: Om368/Fine tuned_CodeT5_Model. I also modified the backend Flask server to load and serve predictions from the fine-tuned model. Additionally, I updated our model documentation to cover model training steps, hyperparameter tuning details, evaluation strategies, and recommendations for future improvements with larger datasets. 

**Why was it changed?**

These changes were made to integrate and deploy our custom-trained model that specializes in generating R code for bioinformatics, enhancing the model's relevance and prediction accuracy for our specific domain. The documentation updates were made to ensure better control over the model's future updates and retraining. The documentation also aims to ensure that future team members can easily retrain the model with more data or improved hyperparameters.

**How was it changed?**

I updated the model loading code within model.py so that it loads our custom fine-tuned model rather than the original base model. The change was specifically here: model = pipeline("text2text-generation", model="Om368/Fine-tuned_CodeT5_Model"). I verified the /predict API still functions correctly by sending test inputs and confirming the outputs. The outputs are definitely better now but need much more improvements in quality and accuracy. I also created a markdown file that outlines everything I know about model training and tuning, as well as future suggestions for improvements.

**Screenshots that show the changes (if applicable):**
image

### CODE REVIEW

**Review by AjithAkuthota23 — APPROVED** [review]
Good work
